### PR TITLE
Skip further lookup work in constraint condensation if current matrix entry is zero.

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: src/Dofs/ConstraintHandler.jl [hunk 1]
```diff
@@ -745,11 +745,12 @@ function _condense!(K::SparseMatrixCSC, f::AbstractVector, dofcoefficients::Vect
         col_coeffs = coefficients_for_dof(dofmapping, dofcoefficients, col)
         if col_coeffs === nothing
             for a in nzrange(K, col)
+                Kval = K.nzval[a]
+                iszero(Kval) && continue
```
Comment (lijas) [inline src/Dofs/ConstraintHandler.jl:749]: Kval will most likely not be zero, right? So maybe it is an unnecessary check?

Comment (fredrikekre) [reply src/Dofs/ConstraintHandler.jl:749]: I don't think it is that uncommon? I.e. if you have no coupling between some dofs?

Comment (lijas) [reply src/Dofs/ConstraintHandler.jl:749]: From the heat_equation tutorial, the stiffness matrix has no zero values (before we call apply!) i.e. `count(i-> i==0, K.nzval) == 0` . But idk, maybe it has many zeros it in other cases?

Comment (KristofferC) [reply src/Dofs/ConstraintHandler.jl:749]: That has only one field. But if you have multiple fields it's often that some fields don't couple in the weak form.

**Pull request merged**